enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1344 @@
+called Maintenance Schedules which define reoccurring windows of time (and specifically
+excluded times) in which potentially disruptive changes in configuration can be initiated. 
+
+Material changes not permitted by change management configuration are left in a 
+pending state until such time as they are permitted by the configuration. 
+
+Change management enforcement _does not_ guarantee that all initiated
+material changes are completed by the close of a permitted change window (e.g. a worker-node
+may still be draining or rebooting) at the close of a scheduled permissive window, 
+but it does prevent _additional_ material changes from being initiated. 
+
+A "material change" may vary by cluster profile and subsystem. For example, a 
+control-plane update (all components and control-plane nodes updated) is implemented as
+a single material change (e.g. the close of a scheduled permissive window
+will not suspend its progress). In contrast, the rollout of worker-node updates is
+more granular (you can consider it as many individual material changes) and  
+the end of a permitted change window will prevent additional worker-node updates 
+from being initiated.
+
+Changes vital to the continued operation of the cluster (e.g. certificate rotation) 
+are not considered material changes. Ignoring operational practicalities (e.g.
+the need to fix critical bugs or update a cluster to supported software versions), 
+it should be possible to safely leave changes pending indefinitely. That said,
+Service Delivery and/or higher level management systems may choose to prevent
+such problematic change management settings from being applied by using 
+validating webhooks or admission policies.
+
+## Definitions and Reference
+
+**RRULE**
+RRULE, or "Recurrence Rule", is an RFC https://icalendar.org/RFC-Specifications/iCalendar-RFC-5545/
+commonly used to express reoccurring windows of time. Consider a calendar invite for a meeting that
+should occur on the last Friday of every month. RRULE can express this as `FREQ=MONTHLY;INTERVAL=1;BYDAY=-1FR`.
+While commonly employed for calendar data interchange, it is used in this enhancement to allow users
+the ability to specify maintenance schedules.
+Tools for generating RRULES:
+- Simple: https://icalendar.org/rrule-tool.html
+- Complex: https://exontrol.com/exicalendar.jsp?config=/js#calendar
+
+**Change Management Terminology**
+This document uses specialized terms to describe the key aspects of change management. 
+It is worth internalizing the meaning of these terms before reviewing sections of the document.
+- "Material Change". A longer definition is provided in the Summary, but, in short, any configuration
+  change a platform operator wishes to apply which would necessitate the reboot or replacement of one 
+  or more nodes is considered a material change. For example, updating the control-plane version is 
+  a material change as it requires rebooting master nodes. 

```

Comment:
Is there a differentiation between standalone and HCP needed here? If I understand correctly, a control-plane upgrade on a HCP would not require rebooting nodes specifically, but the pods for that HCP. Or am I just splitting unnecessary hairs here?

Reply:
I'd consider control plane updates to be a material change whether or not it's standalone or HCP, not because pods or nodes are restarted, but because new code or config gets deployed and behavior will change. That's why we seek to be able to confine HostedControlPlane, ClusterVersion, MachineConfigPool, and NodePool updates to the permissive periods in a given schedule.

Probably splitting hairs too, but would a rolling reboot of nodes with no change be considered a material change? Maybe not for control plane nodes where things are well tuned for zero disruption, including CI tests that monitor for disruption, but user workloads are often less tolerant of rolling restarts. Either way, if initiated via `oc adm reboot-machine-config-pool` they'd both be confined to permissive windows because this actually applies a benign MachineConfig change and the MCO would observe any defined schedule.